### DISCUSSION TITLE
Implement VarcharArray DecodeText in pgx/v5

### DISCUSSION THREAD

**Discussion by mvalletta** [jackc/pgx#2158] [Q&A]
Hi,
I'm trying to parse a varchar array column from wal2json output plugin.
This JSON coming from plugin
```json
{
  "nextlsn": "3/A9856F98",
  "change": [
    {
      "kind": "insert",
      "schema": "public",
      "table": "label",
      "columnnames": [
        "labels"
      ],
      "columntypes": [
        "text[]"
      ],
      "columnvalues": [
        "{aaa bbb ccc}"
      ]
    }
  ]
}
```
With pgx/v3 this code works fine
```go
labels := pgtype.VarcharArray{}
err := labels.DecodeText(pgConnectionInfo, []byte(w.Columnvalues[0].(string)))
if err != nil {
  panic("decoding varchar array")
}
var ll []string
if labels.Status == pgtype.Present {
  err = labels.AssignTo(&ll)
  if err != nil {
    panic"assigning labels")
  }
}
```
With pgx/v5 I can't found a solution
Thanks for your help

**Comment by jackc** [comment]
In pgx v5 you can scan directly into a `[]string` or use `pgtype.Array[string]` or `pgtype.FlatArray[string]`. If you are scanning outside of a query then https://pkg.go.dev/github.com/jackc/pgx/v5@v5.7.1/pgtype#Map.Scan may do what you want.
